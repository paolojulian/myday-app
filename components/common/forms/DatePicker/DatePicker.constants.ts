export const DAY_OF_WEEK = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'];
export const DATE_PICKER_TEST_IDS = {
  container: 'date-picker-container',
  headerBtn: 'date-picker-header-btn',
  value: 'date-picker-value',
  calendar: 'date-picker-calendar',
};
