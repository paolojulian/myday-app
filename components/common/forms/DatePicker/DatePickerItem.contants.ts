export const DATE_PICKER_ITEM_TEST_IDS = {
  containerBtn: (id: string) => `date-picker-item-container-${id}`,
};
