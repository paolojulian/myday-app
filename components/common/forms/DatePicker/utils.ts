import dayjs from 'dayjs';

/**
 * Returns a 2D array of calendar days for a given year and month.
 * Each element in the array represents a week, and each week contains objects representing individual days.
 * The objects contain the date and a flag indicating whether the day belongs to the current month.
 *
 * @param year - The year of the calendar.
 * @param month - The month of the calendar (1-12).
 * @returns A 2D array of calendar days.
 */
export function getCalendarDays(
  year: number,
  month: number,
): { date: Date; isCurrentMonth: boolean }[][] {
  const firstDayOfMonth = new Date(year, month - 1, 1).getDay();
  const daysInMonth = new Date(year, month, 0).getDate();
  const weeksArray = [];
  let week = [];

  // Add previous month days
  const previousMonth = month === 1 ? 12 : month - 1;
  const previousMonthYear = month === 1 ? year - 1 : year;
  const daysInPreviousMonth = new Date(previousMonthYear, previousMonth, 0).getDate();
  const startDay = daysInPreviousMonth - firstDayOfMonth + 1;
  for (let i = startDay; i <= daysInPreviousMonth; i++) {
    week.push({ date: new Date(previousMonthYear, previousMonth - 1, i), isCurrentMonth: false });
    if (week.length === 7) {
      weeksArray.push(week);
      week = [];
    }
  }

  // Add current month days
  for (let i = 1; i <= daysInMonth; i++) {
    week.push({ date: new Date(year, month - 1, i), isCurrentMonth: true });
    if (week.length === 7) {
      weeksArray.push(week);
      week = [];
    }
  }

  // Add next month days
  const nextMonth = month === 12 ? 1 : month + 1;
  const nextMonthYear = month === 12 ? year + 1 : year;
  const remainingDays = 7 - (week.length % 7);
  for (let i = 1; i <= remainingDays; i++) {
    week.push({ date: new Date(nextMonthYear, nextMonth - 1, i), isCurrentMonth: false });
  }
  if (week.length > 0) {
    weeksArray.push(week);
  }

  return weeksArray;
}

export function isToday(date: Date) {
  return dayjs().isSame(date, 'day');
}

export function getCalendarTitle(date?: Date) {
  if (!date) {
    return 'Select a date';
  }

  if (date && isToday(date)) {
    const formattedDateToday = dayjs(date).format('MMM D,YYYY');
    return `Today, ${formattedDateToday}`;
  }

  const tomorrow = dayjs().add(1, 'day');
  if (date && dayjs(date).isSame(tomorrow, 'day')) {
    const formattedDateTomorrow = dayjs(date).format('MMM D,YYYY');
    return `Tomorrow, ${formattedDateTomorrow}`;
  }

  const valueDayJS = dayjs(date);
  return valueDayJS.format('ddd, MMM D, YYYY');
}
