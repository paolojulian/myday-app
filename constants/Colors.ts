export const colors = {
  aliceBlue: '#E5ECF3',
  backdrop: 'rgba(0, 0, 0, 0.3)',
  black: '#05141F',
  darkGrey: '#737373',
  green: '#009000',
  grey: '#a3a3a3',
  lightGrey: '#f8fafc',
  red: '#CC3333',
  white: '#FFFFFF',
  whiteSmoke: '#f5f5f5',
  primary: {
    100: '#F8F9FA',
    200: '#E9ECEF',
    300: '#DEE2E6',
    400: '#CED4DA',
    500: '#ADB5BD',
    600: '#6C757D',
    700: '#495057',
    800: '#343A40',
    900: '#212529',
  },
  danger: '#DC3545',
};
