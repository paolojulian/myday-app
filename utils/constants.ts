export enum TabName {
  Home = 'index',
  Journal = 'journal',
  Expense = 'expenses',
  Todo = 'todos',
}
