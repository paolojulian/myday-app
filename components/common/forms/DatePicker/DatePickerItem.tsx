import { DATE_PICKER_ITEM_TEST_IDS } from '@/components/common/forms/DatePicker/DatePickerItem.contants';
import ThemedText from '@/components/common/ThemedText';
import ThemedView from '@/components/common/ThemedView';
import { colors } from '@/constants/Colors';
import dayjs from 'dayjs';
import React from 'react';
import { StyleProp, StyleSheet, TextStyle, TouchableOpacity, ViewStyle } from 'react-native';

type DatePickerItemProps = {
  onPress?: (date: Date) => void;
  value: Date | string;
  isActive?: boolean;
  isToday?: boolean;
  isCurrentMonth?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
};

function DatePickerItem({
  onPress,
  value,
  isActive = false,
  isToday = false,
  isCurrentMonth = true,
  containerStyle = {},
  textStyle = {},
}: DatePickerItemProps) {
  const handlePress = () => {
    if (typeof value === 'string') {
      return;
    }

    if (onPress) {
      onPress(value);
    }
  };

  return (
    <TouchableOpacity
      onPress={handlePress}
      testID={DATE_PICKER_ITEM_TEST_IDS.containerBtn(
        typeof value === 'string' ? value : dayjs(value).format('YYYY-MM-DD'),
      )}
    >
      <ThemedView
        style={[
          styles.container,
          {
            ...(isToday && styles.isToday),
            ...(isActive && styles.isActive),
          },
          containerStyle,
        ]}
      >
        <ThemedText
          style={[
            {
              ...(isActive && styles.isActiveText),
              ...(!isCurrentMonth && styles.isNotCurrentMonth),
            },
            textStyle,
          ]}
          variant="body1"
        >
          {typeof value === 'string' ? value : value.getDate()}
        </ThemedText>
      </ThemedView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    width: 40,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 9999,
  },
  isActive: {
    backgroundColor: colors.black,
  },
  isToday: {
    backgroundColor: colors.aliceBlue,
  },
  isActiveText: {
    color: colors.white,
  },
  isNotCurrentMonth: {
    color: colors.grey,
  },
});

export default DatePickerItem;
