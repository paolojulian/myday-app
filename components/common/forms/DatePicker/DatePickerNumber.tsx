import ThemedText from '@/components/common/ThemedText';
import ThemedView from '@/components/common/ThemedView';
import { colors } from '@/constants/Colors';
import React from 'react';
import { StyleSheet } from 'react-native';

type DatePickerNumberProps = {
  value: string | number;
  isActive: boolean;
  isToday: boolean;
  isCurrentMonth?: boolean;
};

function DatePickerNumber({
  value,
  isActive,
  isToday,
  isCurrentMonth = true,
}: DatePickerNumberProps) {
  return (
    <ThemedView
      style={[styles.container, isActive ? styles.isActive : {}, isToday ? styles.isToday : {}]}
    >
      <ThemedText
        style={{
          ...(isActive && styles.isActiveText),
          ...(!isCurrentMonth && styles.isNotCurrentMonth),
        }}
      >
        {value}
      </ThemedText>
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    width: 32,
    height: 32,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 9999,
  },
  isActive: {
    backgroundColor: colors.black,
  },
  isToday: {
    backgroundColor: colors.aliceBlue,
  },
  isActiveText: {
    color: colors.white,
  },
  isNotCurrentMonth: {
    color: colors.grey,
  },
});

export default DatePickerNumber;
