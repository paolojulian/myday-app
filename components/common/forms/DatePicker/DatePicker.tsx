import {
  DAY_OF_WEEK,
  DATE_PICKER_TEST_IDS,
} from '@/components/common/forms/DatePicker/DatePicker.constants';
import DatePickerItem from '@/components/common/forms/DatePicker/DatePickerItem';
import {
  getCalendarDays,
  getCalendarTitle,
  isToday,
} from '@/components/common/forms/DatePicker/utils';
import Label from '@/components/common/forms/Label';
import ThemedText from '@/components/common/ThemedText';
import ThemedView from '@/components/common/ThemedView';
import { colors } from '@/constants/Colors';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import dayjs from 'dayjs';
import React, { useMemo, useState } from 'react';
import { Image, StyleSheet, TouchableWithoutFeedback } from 'react-native';

type DatePickerVariants = 'shadow' | 'border';

type DatePickerProps = {
  onSelectDate?: (date: Date) => void;
  canShrink?: boolean;
  initialIsExpanded?: boolean;
  initialMonth?: number;
  initialYear?: number;
  label?: string;
  value?: Date;
  variant?: DatePickerVariants;
};

function DatePicker({
  onSelectDate,
  canShrink = true,
  initialIsExpanded = false,
  initialMonth,
  initialYear,
  label = 'Date',
  value,
  variant = 'border',
}: DatePickerProps) {
  const [year, setYear] = useState<number>(initialYear ?? new Date().getFullYear());
  const [month, setMonth] = useState<number>(initialMonth ?? new Date().getMonth() + 1);
  const [isExpanded, setIsExpanded] = useState<boolean>(initialIsExpanded);

  const calendarDays = useMemo(() => {
    return getCalendarDays(year, month);
  }, [year, month]);

  const isDayActive = (date: Date) => {
    return dayjs(value).isSame(date, 'day');
  };

  const titleText = useMemo(() => {
    return getCalendarTitle(value);
  }, [value]);
  const currentMonthTitle = useMemo(() => {
    const monthTitle = dayjs(new Date(year, month - 1)).format('MMM');

    const currentYear = new Date().getFullYear();
    if (year !== currentYear) {
      return `${monthTitle} ${year}`;
    } else {
      return monthTitle;
    }
  }, [month, year]);

  const handleHeaderClick = () => {
    if (canShrink) {
      setIsExpanded(prev => !prev);
    } else {
      setIsExpanded(true);
    }
  };

  const handlePrevMonth = () => {
    if (month === 1) {
      setMonth(12);
      setYear(year - 1);
    } else {
      setMonth(month - 1);
    }
  };

  const handleNextMonth = () => {
    if (month === 12) {
      setMonth(1);
      setYear(year + 1);
    } else {
      setMonth(month + 1);
    }
  };

  return (
    <ThemedView
      testID={DATE_PICKER_TEST_IDS.container}
      style={[
        styles.container,
        {
          ...(variant === 'shadow' && styles.containerShadow),
          ...(variant === 'border' && styles.containerBorder),
        },
      ]}
    >
      <ThemedView style={styles.header}>
        <TouchableWithoutFeedback
          testID={DATE_PICKER_TEST_IDS.headerBtn}
          onPress={handleHeaderClick}
        >
          <ThemedView style={{ flex: 1, flexDirection: 'row', gap: 16, alignItems: 'center' }}>
            <Image
              source={require('@/assets/icons/calendar-lines.png')}
              style={{ width: 24, height: 24 }}
            />
            <ThemedView style={{ gap: 4, flex: 1 }}>
              <Label text={label} />
              <ThemedText
                testID={DATE_PICKER_TEST_IDS.value}
                variant="body1"
                style={{ color: !!value ? colors.black : colors.grey }}
              >
                {titleText}
              </ThemedText>
            </ThemedView>
          </ThemedView>
        </TouchableWithoutFeedback>
        {isExpanded && (
          <ThemedView style={styles.monthNavigation}>
            <MaterialCommunityIcons
              size={40}
              name="chevron-left"
              color={colors.black}
              onPress={handlePrevMonth}
            />
            <ThemedText style={{ width: 40, textAlign: 'center' }} variant="body1">
              {currentMonthTitle}
            </ThemedText>
            <MaterialCommunityIcons
              size={40}
              name="chevron-right"
              color={colors.black}
              onPress={handleNextMonth}
            />
          </ThemedView>
        )}
      </ThemedView>
      {isExpanded && (
        <ThemedView style={styles.body} testID={DATE_PICKER_TEST_IDS.calendar}>
          <ThemedView style={styles.weekContainer}>
            {DAY_OF_WEEK.map(day => (
              <DatePickerItem
                key={day}
                value={day}
                textStyle={{ color: colors.grey, fontSize: 12 }}
              />
            ))}
          </ThemedView>
          {calendarDays.map((week, index) => (
            <ThemedView key={index} style={styles.weekContainer}>
              {week.map(({ date, isCurrentMonth }) => {
                return (
                  <DatePickerItem
                    onPress={onSelectDate}
                    key={date.toString()}
                    value={date}
                    isActive={value ? isDayActive(date) : false}
                    isToday={isToday(date)}
                    isCurrentMonth={isCurrentMonth}
                  />
                );
              })}
            </ThemedView>
          ))}
        </ThemedView>
      )}
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {},
  containerShadow: {
    shadowColor: colors.black,
    shadowOpacity: 0.24,
    shadowRadius: 12,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    elevation: 5,
  },
  containerBorder: {
    borderWidth: 2,
    borderRadius: 12,
    borderColor: colors.whiteSmoke,
  },
  header: {
    borderTopStartRadius: 12,
    borderTopEndRadius: 12,
    backgroundColor: colors.whiteSmoke,
    paddingVertical: 16,
    paddingHorizontal: 16,

    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-end',
    justifyContent: 'space-between',
  },
  monthNavigation: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  body: {
    borderBottomStartRadius: 12,
    borderBottomEndRadius: 12,
    backgroundColor: colors.white,
    padding: 16,
  },
  weekContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
});

export default DatePicker;
