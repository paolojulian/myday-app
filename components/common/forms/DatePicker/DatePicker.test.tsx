import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import DatePicker from '@/components/common/forms/DatePicker/DatePicker';
import { DATE_PICKER_TEST_IDS } from '@/components/common/forms/DatePicker/DatePicker.constants';
import { DATE_PICKER_ITEM_TEST_IDS } from '@/components/common/forms/DatePicker/DatePickerItem.contants';
import dayjs from 'dayjs';

describe('TESTING DatePicker', () => {
  describe('WHEN the component is rendered', () => {
    it('should match snapshot', () => {
      const component = render(<DatePicker />);
      expect(component).toMatchSnapshot();
    });
    it('renders without crashing', () => {
      render(<DatePicker />);
    });
    it('displays the label correctly', () => {
      const { getByText, rerender } = render(<DatePicker label="Select Date" />);
      expect(getByText('Select Date')).toBeTruthy();

      rerender(<DatePicker label="New Label" />);
      expect(getByText('New Label')).toBeTruthy();
    });
  });

  describe('WHEN the date picker is expanded', () => {
    it('should display the current month', () => {
      render(<DatePicker />);
      expandDatePicker();
      expect(screen.getByText(dayjs().format('MMM'))).toBeTruthy();
    });
  });

  describe('WHEN a date is selected', () => {
    it('calls onSelectDate when a date is selected', () => {
      const handleSelectDate = jest.fn();
      const { rerender } = render(<DatePicker onSelectDate={handleSelectDate} />);
      expandDatePicker();
      // Simulate selecting a date
      const dateToSelect = dayjs();
      fireEvent.press(
        screen.getByTestId(
          DATE_PICKER_ITEM_TEST_IDS.containerBtn(dateToSelect.format('YYYY-MM-DD')),
        ),
      );
      expect(handleSelectDate).toHaveBeenCalled();

      // should display the selected date
      rerender(<DatePicker value={dateToSelect.toDate()} />);
      expect(screen.getByTestId(DATE_PICKER_TEST_IDS.value).children[0]).toBe(
        `Today, ${dateToSelect.format('MMM D,YYYY')}`,
      );
    });
  });

  describe('WHEN there is no date selected', () => {
    it('should not display a selected date', () => {
      render(<DatePicker value={undefined} />);
      expandDatePicker();

      const today = dayjs().format('YYYY-MM-DD');
      const isActive = false;
      expect(screen.getByTestId(DATE_PICKER_ITEM_TEST_IDS.isActive(today, isActive)));
    });
  });

  describe('WHEN the date picker is collapsed', () => {
    it('should hide the calendar', () => {
      render(<DatePicker />);
      expandDatePicker();
      expect(screen.getByTestId(DATE_PICKER_TEST_IDS.calendar)).toBeTruthy();
      collapseDatePicker();
      expect(screen.queryByTestId(DATE_PICKER_TEST_IDS.calendar)).toBeNull();
    });
  });
});

function expandDatePicker() {
  fireEvent.press(screen.getByTestId(DATE_PICKER_TEST_IDS.headerBtn));
}

function collapseDatePicker() {
  fireEvent.press(screen.getByTestId(DATE_PICKER_TEST_IDS.headerBtn));
}
