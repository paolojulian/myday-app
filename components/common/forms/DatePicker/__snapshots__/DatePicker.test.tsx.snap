// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING DatePicker WHEN the component is rendered should match snapshot 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "borderRadius": 0,
      },
    ]
  }
  testID="date-picker-container"
>
  <View
    style={
      {
        "alignItems": "flex-end",
        "backgroundColor": "#404040",
        "borderTopEndRadius": 8,
        "borderTopStartRadius": 8,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 8,
        "paddingHorizontal": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "gap": 16,
        }
      }
      testID="date-picker-header-btn"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="25"
        bbWidth="26"
        color="#fafafa"
        fill="none"
        focusable={false}
        height="25"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 25,
              "width": 26,
            },
          ]
        }
        tintColor="#fafafa"
        vbHeight={25}
        vbWidth={26}
        width="26"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M1.72913 22.6562C1.72913 23.9502 2.77893 25 4.07288 25H21.2604C22.5543 25 23.6041 23.9502 23.6041 22.6562V9.375H1.72913V22.6562ZM17.3541 13.0859C17.3541 12.7637 17.6178 12.5 17.9401 12.5H19.8932C20.2155 12.5 20.4791 12.7637 20.4791 13.0859V15.0391C20.4791 15.3613 20.2155 15.625 19.8932 15.625H17.9401C17.6178 15.625 17.3541 15.3613 17.3541 15.0391V13.0859ZM17.3541 19.3359C17.3541 19.0137 17.6178 18.75 17.9401 18.75H19.8932C20.2155 18.75 20.4791 19.0137 20.4791 19.3359V21.2891C20.4791 21.6113 20.2155 21.875 19.8932 21.875H17.9401C17.6178 21.875 17.3541 21.6113 17.3541 21.2891V19.3359ZM11.1041 13.0859C11.1041 12.7637 11.3678 12.5 11.6901 12.5H13.6432C13.9655 12.5 14.2291 12.7637 14.2291 13.0859V15.0391C14.2291 15.3613 13.9655 15.625 13.6432 15.625H11.6901C11.3678 15.625 11.1041 15.3613 11.1041 15.0391V13.0859ZM11.1041 19.3359C11.1041 19.0137 11.3678 18.75 11.6901 18.75H13.6432C13.9655 18.75 14.2291 19.0137 14.2291 19.3359V21.2891C14.2291 21.6113 13.9655 21.875 13.6432 21.875H11.6901C11.3678 21.875 11.1041 21.6113 11.1041 21.2891V19.3359ZM4.85413 13.0859C4.85413 12.7637 5.1178 12.5 5.44006 12.5H7.39319C7.71545 12.5 7.97913 12.7637 7.97913 13.0859V15.0391C7.97913 15.3613 7.71545 15.625 7.39319 15.625H5.44006C5.1178 15.625 4.85413 15.3613 4.85413 15.0391V13.0859ZM4.85413 19.3359C4.85413 19.0137 5.1178 18.75 5.44006 18.75H7.39319C7.71545 18.75 7.97913 19.0137 7.97913 19.3359V21.2891C7.97913 21.6113 7.71545 21.875 7.39319 21.875H5.44006C5.1178 21.875 4.85413 21.6113 4.85413 21.2891V19.3359ZM21.2604 3.125H18.9166V0.78125C18.9166 0.351562 18.5651 0 18.1354 0H16.5729C16.1432 0 15.7916 0.351562 15.7916 0.78125V3.125H9.54163V0.78125C9.54163 0.351562 9.19006 0 8.76038 0H7.19788C6.76819 0 6.41663 0.351562 6.41663 0.78125V3.125H4.07288C2.77893 3.125 1.72913 4.1748 1.72913 5.46875V7.8125H23.6041V5.46875C23.6041 4.1748 22.5543 3.125 21.2604 3.125Z"
            fill={
              {
                "type": 2,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          {
            "flex": 1,
            "gap": 4,
          }
        }
      >
        <Text
          aria-label="Date"
          style={
            [
              {
                "fontFamily": "RalewayBold",
                "fontSize": 16,
                "fontWeight": 700,
              },
              {
                "color": "#fafafa",
              },
              {
                "fontWeight": 400,
              },
              {
                "color": "#fafafa",
              },
            ]
          }
        >
          Date
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "RalewayRegular",
                "fontSize": 16,
              },
              {
                "color": "#fafafa",
              },
              {
                "fontWeight": 400,
              },
              {
                "color": "#A3A3A3",
              },
            ]
          }
          testID="date-picker-value"
        >
          Select a date
        </Text>
      </View>
    </View>
  </View>
</View>
`;
